Classes/ContentRepository/Service/WorkspaceService.php

```diff
@@ -35,6 +34,11 @@
  */
 class WorkspaceService
 {
+    private const NODE_HAS_BEEN_CREATED = 0b0001;
+    private const NODE_HAS_BEEN_CHANGED = 0b0010;
+    private const NODE_HAS_BEEN_MOVED = 0b0100;
+    private const NODE_HAS_BEEN_DELETED = 0b1000;
```

Comment:
that seems clever and cost effective but isnt this a bit hard to grasp as the information is interpreted in the js codbase?

Reply:
In the other PR #3759, I've already refactored the client side part of this: https://github.com/neos/neos-ui/pull/3759/commits/3ce8499159180c8a6ed3e3a39944fe6a5e908024

Bitmasks work in JS just as they do in PHP, so this can simply be mirrored.